Process both tumor and normal BAM files in Mutect2 wrapper

lgpdv opened the pull request:
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

drosofff commented:
Hi @lgpdv 
You should probably, poke the tool developers, if possible request some reviewing, and probably bump the tool version following the iuc rules ;-)

lgpdv commented:
Please @davebx , could you please review the proposed changes?

Thanks in advance,

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0"?>
 <macros>
     <token name="@VERSION@">4.1.4.0</token>
-    <token name="@WRAPPER_VERSION@">@VERSION@+galaxy</token>
```
please leave this in, but change it to `@VERSION@+galaxy1`

lgpdv replied:
Hi @bgruening ,
Thanks a lot for reviewing the proposed changes. On the other side, I've corrected the mentioned mistakes.

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,82 @@
+        #if $varExists($sect + "normal")
+            #if $getVar($sect + "normal").is_of_type("bam")
+                ln -s $getVar($sect + "normal") normal.bam &&
+                samtools index normal.bam &&
```
@lgpdv I don't think the indexing step is needed.
You can use the index Galaxy provides you. E.g. wth `ln -s '${sect.normal.metadata.bam_index}'  normal.bam.bai &&`

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,82 @@
+            #else
+                ln -s $getVar($sect + "normal") normal.sam &&
+                samtools view -bS normal.sam -o normal.bam &&
+                samtools index normal.bam &&
```
same here we can save the index step

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,82 @@
+    <template name="picard_normal_bam_index"><![CDATA[
+        #if $normal.is_of_type("bam")
+            ln -s $INPUT normal.bam &&
+            samtools index normal.bam &&
```
we can remove the index step

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,82 @@
+        #else
+            ln -s $INPUT normal.sam &&
+            samtools view -bS normal.sam -o normal.bam &&
+            samtools index normal.bam &&
```
remove index

lgpdv replied:
Thanks @bgruening for the suggested modifications! In consequence, I have deleted the index step from the code and have replaced it by the symbolic links to corresponding files.

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,82 @@
+        #else
+            ln -s $INPUT normal.sam &&
+            samtools view -bS normal.sam -o normal.bam &&
+            ln -s $INPUT normal.bam.bai &&
```
This line one can be removed.

bgruening commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,82 @@
+            #else
+                ln -s $getVar($sect + "normal") normal.sam &&
+                samtools view -bS normal.sam -o normal.bam &&
+                ln -s '${normal.metadata.bam_index}' normal.bam.bai &&
```
If normal in a SAM file there is no index, so this line can be removed.

bgruening replied:
Also if you like you can completely remove the option to include SAM files ... galaxy can autoconvert SAM to BAM if really needed.

lgpdv replied:
Good to know about these automated tasks in Galaxy! Thanks a lot for amended the code @bgruening

bgruening commented:
Sure thing! Thanks for your contribution this is awesome!

bgruening commented:
@lgpdv I just made the final review and realized there is a newer mutect2 version. Let's see if tests passes as well. I can revert my changes if you like to keep the old version.

bgruening commented:
Uha, ok this upgrade might be bigger than expected. @lgpdv should I revert?

lgpdv commented:
> Uha, ok this upgrade might be bigger than expected. @lgpdv should I revert?

@bgruening yes, indeed this version causes some tests to fail. I am running the tests on my local computer to analyze which are the errors and if they can be amended from the command line. If not, then I will let you know to revert to the previous version.

bgruening commented:
Awesome, thanks!

lgpdv commented:
> Awesome, thanks!

@bgruening I have checked that most of the mistakes arrives from the differences between the old / new output files, that can be updated very easily. However, in 2 cases they arrive from deprecated optional arguments (--activity-profile-out & --dont_trim_active_regions) in the new version. Nevertheless, since 2 tests and their outputs are related to these options, I do not know if the original author of the wrapper @davebx will make use of them for his analysis. If this is the case, the previous version should be kept instead.

bgruening commented:
I think it is safe to update, but I will try to ping Dave.

lgpdv commented:
> I think it is safe to update, but I will try to ping Dave.

@bgruening I have updated the output files and deleted these options from the code and all tests pass. But, as I've mentioned, 2 tests produced some tabular output files that are not created anymore when deleting these options.

bgruening commented:
@lgpdv feel free to remove those outputs. Do you also have time to check if new options have been added?

lgpdv commented:
> @lgpdv feel free to remove those outputs. Do you also have time to check if new options have been added?

I will check about the new options. Regarding the mentioned ones, they have been removed from the version GATK 4.1.5.0. Moreover, some people have asked about some substitutions to these deleted options to the GATK Team, but not answers were given so far.

davebx commented:
@lgpdv I don't use those options or files, Björn's advice is (as usual) solid.

davebx commented on tools/gatk4/macros.xml:
```diff
@@ -290,45 +290,58 @@
     <!--BAM input-->
-    <template name="picard_bam_input">
-        --INPUT input.bam
+    <template name="picard_tumor_bam_input">
```
I think tumor-related options should go in the mutect2 wrapper itself, the macros.xml definitions need to be broadly applicable to other tools in the GATK suite when people have the bandwidth to wrap them.

lgpdv replied:
@davebx  thanks a lot for this suggestion and previous information. I will check about including the tumor-related options in the mutect2 wrapper itself.

lgpdv commented:
> 
> 
> @lgpdv feel free to remove those outputs. Do you also have time to check if new options have been added?

@bgruening I have checked that only a new option has been added (--allele-informative-reads-overlap-margin) to this new version of the tool. Therefore, along with the mentioned changes concerning deprecated options, I have included it in the new commit of the wrapper.

bgruening commented:
Thanks a lot @lgpdv! This is great work! Welcome to IUC :)

lgpdv commented:
> 
> 
> Thanks a lot @lgpdv! This is great work! Welcome to IUC :)

Thank-you @bgruening for the review & suggestions + corrections!
